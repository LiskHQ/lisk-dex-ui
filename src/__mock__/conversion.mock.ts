export const mockConversionRates = {
  LSK: {
    USD: 0.8863,
    DEX: 0.1,
  },
  DEX: {
    USD: 0.0886
  }
};