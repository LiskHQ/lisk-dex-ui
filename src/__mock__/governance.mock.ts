import { ProposalType } from "consts";
import { IPoolItem, IProposal } from "models";

export const mockProposal: IProposal = {
  author: "Author",
  proposalType: ProposalType.PoolIncentivization,
  poolID: "1",
  multiplier: 8,
  title: "title",
  summary: "summary",
  description: "description",
  link: "description"
}

export const mockPoolItems: IPoolItem[] = [
  {
    id: '1',
    chain1: 'LSK',
    chain2: 'DEU',
    rate: 0.3,
    amount: 12.2,
  },
  {
    id: '2',
    chain1: 'KLP',
    chain2: 'RGB',
    rate: 0.1,
    amount: 11.4,
  },
  {
    id: '3',
    chain1: 'DEU',
    chain2: 'FAE',
    rate: 0.3,
    amount: 17.6,
  },
  {
    id: '4',
    chain1: 'RGB',
    chain2: 'FAE',
    rate: 0.1,
    amount: 11.4,
  }
]