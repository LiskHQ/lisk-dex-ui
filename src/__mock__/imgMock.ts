export const svgUrl: string = 'svg-url';