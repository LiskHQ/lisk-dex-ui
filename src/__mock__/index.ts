export * from './governance.mock';
export * from './user.mock';