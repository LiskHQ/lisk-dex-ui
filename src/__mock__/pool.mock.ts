import { IPool } from "models";
import { mockTokens } from "./swap.mock";

export const mockPool: IPool = {
  token1: mockTokens[0],
  token2: mockTokens[1],
  token1Amount: 1862,
  token2Amount: 0.045,
  share: 0.085,
}

export const mockPools: IPool[] = [
  mockPool,
]