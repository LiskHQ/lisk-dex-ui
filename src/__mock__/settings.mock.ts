import { flagSvgs } from 'imgs/icons';
import { IFiatCurrency } from 'models';

export const mockFiatCurrnecies: IFiatCurrency[] = [
  {
    symbol: 'USD',
    name: 'US Dollar',
    image: flagSvgs.US,
  },
  {
    symbol: 'EUR',
    name: 'EU Euro',
    image: flagSvgs.EU,
  }
];