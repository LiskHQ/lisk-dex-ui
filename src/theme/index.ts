export { default as theme } from "./theme";
export { default as variables } from "./variables";
