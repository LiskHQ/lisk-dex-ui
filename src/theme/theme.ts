import {
  createTheme,
  responsiveFontSizes,
  ThemeOptions,
} from "@mui/material/styles";

import variables from "./variables";

const options: ThemeOptions = {
  palette: {
    primary: variables.primary,
    secondary: variables.secondary,
  },
  typography: {
    h1: {
      fontSize: "2.5rem",
      fontWeight: 600,
    },
    h2: {
      fontSize: "2rem",
      fontWeight: 400,
    },
    h3: {
      fontSize: "1.75rem",
      fontWeight: 400,
    },
    h4: {
      fontSize: "1.25rem",
      fontWeight: 400,
    },
    h5: {
      fontSize: "1rem",
      fontWeight: 400,
    },
    h6: {
      fontSize: "0.75rem",
      fontWeight: 400,
    },
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: 576,
      md: 768,
      lg: 1024,
      xl: 1440,
      xxl: 1950,
      xxxl: 2300,
    },
  },
};

const theme = responsiveFontSizes(createTheme(options));
export default theme;
