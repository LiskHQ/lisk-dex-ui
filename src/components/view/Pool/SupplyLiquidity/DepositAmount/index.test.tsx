import React from "react";
import { ThemeProvider } from "@mui/material";
import { fireEvent, render } from "@testing-library/react";
import { DepositAmount, IDepositAmountProps } from "./index";
import { lightTheme } from "styles/theme";
import { mockTokens } from "__mock__";

function renderComponent(props: IDepositAmountProps) {
  return render(
    <ThemeProvider theme={lightTheme} >
      <DepositAmount {...props} />
    </ThemeProvider>
  );
}

describe("DepositAmount component", () => {
  const mockProps: IDepositAmountProps = {
    balance: 4521.425,
    tokenAmount: 0,
    token: mockTokens[0],
    onChange: jest.fn(),
  }

  it("checks if the component matches the snapshot", () => {
    const { container } = renderComponent(mockProps);
    expect(container).toMatchSnapshot();
  });

  it("select amount works", () => {
    const { getByDisplayValue, getByTestId } = renderComponent(mockProps);

    fireEvent.click(getByTestId("LSK-amount-percent-max"));

    expect(getByDisplayValue(mockProps.balance)).toBeInTheDocument();
    expect(mockProps.onChange).toBeCalled();
  });
});
