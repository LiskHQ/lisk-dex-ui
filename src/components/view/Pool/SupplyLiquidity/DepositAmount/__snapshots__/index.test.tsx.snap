// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DepositAmount component checks if the component matches the snapshot 1`] = `
.emotion-0 {
  padding: 1rem;
  background: #FAFAFC;
  border: 1px solid #EEF0F8;
  border-radius: 0.5rem;
}

.emotion-0 .token-amount {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
}

.emotion-0 .token-amount .selected-token {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-0 .token-amount .selected-token h6 {
  margin-left: 0.5rem;
}

.emotion-0 .token-amount .select-tokens {
  border-radius: 0.5rem;
  background: linear-gradient(180deg, #513AE2 0%, rgba(64, 45, 183, 0.9) 100%);
  padding: 0.5rem 0.625rem;
  color: #FFFFFF;
}

.emotion-0 .token-amount .MuiInputBase-root {
  border: none;
  padding: 0;
}

.emotion-0 .token-amount .MuiInputBase-root input {
  font-size: 1.25rem;
  text-align: right;
}

.emotion-0 .token-amount .MuiInputBase-root.hover {
  border: none;
}

.emotion-0 .token-amount .MuiInputBase-root.active {
  border: none;
}

.emotion-0 .token-balance-details {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  color: #6953F4;
  margin-top: 1rem;
}

.emotion-0 .token-balance-details .token-balance-percent {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
}

.emotion-0 .token-balance-details .token-balance-percent p {
  margin-left: 0.75rem;
}

.emotion-3 {
  margin: 0;
  font-size: 1.25rem;
  line-height: 1.5rem;
  font-weight: 400;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  letter-spacing: 0.00938em;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
  position: relative;
}

.emotion-4 .MuiInputLabel-root {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: unset;
  color: #282828!important;
  font-size: 0.875rem;
  line-height: 1.124rem;
  font-weight: 400;
  -webkit-transform: none;
  -moz-transform: none;
  -ms-transform: none;
  transform: none;
  -webkit-transform: none;
  -ms-transform: none;
  margin-bottom: 0.5rem;
}

.emotion-4 .MuiInputLabel-root svg {
  cursor: pointer;
  margin-left: 0.25rem;
}

.emotion-4 .MuiInputLabel-root svg path {
  stroke: #6B7280!important;
}

.emotion-4 .MuiInputBase-root {
  padding: 0.75rem 1rem;
  border-radius: 0.25rem;
  color: #282828;
  border: 0.5px solid #E6E6E6;
  background: unset;
  -webkit-animation-duration: 1s!important;
  animation-duration: 1s!important;
  font-size: 1rem;
  line-height: 1.625rem;
}

.emotion-4 .MuiInputBase-root input {
  padding: 0;
}

.emotion-4 .MuiInputBase-root fieldset {
  border: none;
}

.emotion-4 .MuiInputBase-root:before {
  content: none;
}

.emotion-4 .MuiInputBase-root:after {
  content: none;
}

.emotion-4 .MuiInputBase-root:hover {
  border: 1px solid #A196F6;
}

.emotion-4 .MuiInputBase-root:focus-within {
  border: 1px solid #6953F4;
}

.emotion-4 .MuiFormHelperText-root {
  color: #FF4557;
  margin: 0.5rem 0 0 0;
}

.emotion-4 .input-length-counter {
  position: absolute;
  bottom: -1.5rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: end;
  justify-content: end;
  color: #6B7280;
  margin-top: 0.25rem;
  margin-bottom: 0.25rem;
}

.emotion-4 input::-webkit-outer-spin-button,
.emotion-4 input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  display: none;
}

.emotion-5 {
  font-size: 1rem;
  line-height: 1.4375em;
  font-weight: 500;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  letter-spacing: 0.00938em;
  color: rgba(0, 0, 0, 0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
  cursor: default;
}

.emotion-6 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
}

.emotion-6::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-6::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-6:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-6::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-6:focus {
  outline: 0;
}

.emotion-6:invalid {
  box-shadow: none;
}

.emotion-6::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-6::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-6::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-6:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-6::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-6:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-6:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-6:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-6:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-6.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(0, 0, 0, 0.38);
}

.emotion-6:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-6::-webkit-input-placeholder {
  opacity: 1;
  color: #6B7280;
}

.emotion-6::-moz-placeholder {
  opacity: 1;
  color: #6B7280;
}

.emotion-6:-ms-input-placeholder {
  opacity: 1;
  color: #6B7280;
}

.emotion-6::placeholder {
  opacity: 1;
  color: #6B7280;
}

.emotion-8 {
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.125rem;
  font-weight: 400;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  letter-spacing: 0.01071em;
}

<div>
  <div
    class="deposit-amount emotion-0"
  >
    <div
      class="token-amount MuiBox-root emotion-1"
    >
      <div
        class="selected-token MuiBox-root emotion-1"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2728%27%20height=%2728%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 emotion-3"
        >
          LSK
        </h6>
      </div>
      <div
        class="MuiFormControl-root emotion-4"
      >
        <div
          class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl emotion-5"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input emotion-6"
            type="number"
            value="0"
          />
        </div>
      </div>
    </div>
    <div
      class="token-balance-details MuiBox-root emotion-1"
    >
      <p
        class="MuiTypography-root MuiTypography-body2 emotion-8"
      >
        Balance: 
        4521.425
      </p>
      <div
        class="token-balance-percent token MuiBox-root emotion-1"
      >
        <p
          class="MuiTypography-root MuiTypography-body2 emotion-8"
          data-testid="LSK-amount-percent-25"
        >
          25%
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 emotion-8"
          data-testid="LSK-amount-percent-50"
        >
          50%
        </p>
        <p
          class="MuiTypography-root MuiTypography-body2 emotion-8"
          data-testid="LSK-amount-percent-max"
        >
          MAX
        </p>
      </div>
    </div>
  </div>
</div>
`;
